src/k2/module/k23si/Module.cpp

```diff
@@ -472,105 +472,45 @@ std::size_t K23SIPartitionModule::_findField(const dto::Schema schema, k2::Strin
     return fieldNumber;
 }
 
-bool K23SIPartitionModule::_advancePayloadPosition(Payload& payload, dto::FieldType type) {
-    switch (type) {
-    case k2::dto::FieldType::STRING : {
-        k2::String value;
-        bool success = payload.read(value);
-        if (!success) {
-            return false;
-        }
-        break;
-    }
-    case k2::dto::FieldType::INT16T : {
-        int16_t value;
-        bool success = payload.read(value);
-        if (!success) {
-            return false;
-        }
-        break;
-    }
-    case k2::dto::FieldType::INT32T : {
-        int32_t value;
-        bool success = payload.read(value);
-        if (!success) {
-            return false;
-        }
-        break;
-    }
-    case k2::dto::FieldType::INT64T : {
-        int64_t value;
-        bool success = payload.read(value);
-        if (!success) {
-            return false;
-        }
-        break;
-    }
-    case k2::dto::FieldType::FLOAT : {
-        float value;
-        bool success = payload.read(value);
-        if (!success) {
-            return false;
-        }
-        break;
-    }
-    default :
-        return false;
-    }// end switch
-    return true;
+template <typename T>
+void _advancePayloadPosition(const dto::SchemaField& field, Payload& payload, bool& success) {
+    (void) field;
+    T value{};
+    success = payload.read(value);
```

Comment:
I wonder if we should add a `skip<T>()` method to payload so that we don't have to copy bytes that aren't needed if we want to skip ahead